src/inspector/models/notation/chordsymbols/chordsymbolsettingsmodel.cpp

```diff
@@ -59,6 +60,20 @@ void ChordSymbolSettingsModel::resetProperties()
     m_durationType->resetToDefault();
 }
 
+void ChordSymbolSettingsModel::setChordStylesModel(ChordSymbolStylesModel* chordSymbolStylesModel)
+{
+    if (m_chordStylesModel == chordSymbolStylesModel) {
```

Comment:
Seems a bit unnecessary unless you're triggering an event, also I'd generally write setters/getters as inline functions in the header file unless there's a particular reason not to.

Reply:
@wizofaus It's generally our style to put nothing of function definitions in header files.

But I agree the setter is not necessary. Setters like this with such a check are necessary for use with QML, to avoid binding loops. But when a setter is only used in C++, the check is not necessary. In this case, the setter is only used inside the class itself, and only one time (if I'm not mistaken), so I think it should be removed and replaced by just `m_chordStylesModel = new ChordSymbolStylesModel(this);`.

Reply:
Does it need to be alloc'ed on the heap at all? I.e. couldn't m_chordStylesModel just be a ChordSymbolStylesModel instance rather than a pointer?

Reply:
Good point. Also, currently a `delete` is missing, but with just an instance, that would not be necessary. I believe QML can only work with pointers to Object-derived items, so you would need to pass a pointer to the instance in the getter method.

Reply:
Except the current object is passed in as "parent" to the QObject, I'm not sure if that's necessary or what it will mean - if Qt automatically deletes children when you delete the parent it would be an issue!

Reply:
I added the setter method at first thinking I might use it from somewhere else in the code. But now it does seem unnecessary. I will remove it now and add it if I needed it again.
And regarding `ChordSymbolStylesModel` being a pointer or instance, a pointer is fine right? Is there any reason to change it?

Reply:
You just need to check if it automatically gets deleted when the parent does, if not, you'll need to delete it yourself or use an instance.

Reply:
From reading the Qt documentation it seems it will get automatically deleted as you're passing in the current object as the parent.

Reply:
> From reading the Qt documentation it seems it will get automatically deleted as you're passing in the current object as the parent.

Oh Ok.